go-controller/pkg/ovn/namespace.go

```diff
@@ -14,6 +17,27 @@ const (
 	nsMulticastAnnotation = "k8s.ovn.org/multicast-enabled"
 )
 
+func WaitForNamespace(name string, wf *factory.WatchFactory) (*kapi.Namespace, error) {
```

Comment:
@rcarrillocruz Ok, so maybe I led you astray. I now think we should just put this into the WatchFactory as a factory method. IT's kinda ugly to pass the factory in as a parameter, and this function doesn't use anything outside of the factory.

Reply:
Instead of doing this, we already watch namespaces and cache them. We check if the namespace exists when we try to lock it here:
https://github.com/ovn-org/ovn-kubernetes/blob/477f00d358a37a6f788fdcbc41dad1d54b5c3f64/go-controller/pkg/ovn/namespace.go#L41

which happens during the add logical port. Cant we just wait using this information instead of the watch factory?

Reply:
@trozet @rcarrillocruz yes, you could use `waitForNamespaceLocked()`, except that we don't cache namespace annotations which is what Ricky needs to use here. If we started caching that in nsInfo (which would probably be useful) then we could use `waitForNamespaceLocked()`

Reply:
yeah I think we should do that. you agree?

Reply:
ok, I'll see how that looks like and push.

Reply:
Check it out now, I added the namespaces field to namespaceInfo struct and I initialize it to the namespace annotations within createNamespaceLocked.
Should I put the wf.GetNamespace in an ExponentialBackoff? I assum so?
Also, not sure how I will leverage this from node_linux.go , I don't have an oc there to call a method like WaitForNamespaceAnnotations.

Reply:
you don't need this function in ovn/namespace.go (see other comments), but you do need something like it in node_linux.go. So move it over there and then your issue in node_linux.go on line 63 goes away. 

You don't have to use a watch factory if you don't want to, you could just use kubeclient with retry/backoff mechanism. I don't see what the advantage is of one over the other, wf seems a little more heavy for just getting 1 object. @dcbw what do you think?